src/org/mozilla/javascript/NativeString.java

```diff
@@ -242,7 +242,13 @@ public Object execIdCall(IdFunctionObject f, Context cx, Scriptable scope,
                     }
                     int[] codePoints = new int[n];
                     for (int i = 0; i != n; i++) {
-                        codePoints[i] = ScriptRuntime.toInt32(args[i]);
+                        Object arg = args[i];
+                        int codePoint = ScriptRuntime.toInt32(arg);
+                        double num = ScriptRuntime.toNumber(arg);
+                        if (!ScriptRuntime.eqNumber(num, codePoint) || !Character.isValidCodePoint(codePoint)) {
+                            throw rangeError("invalid code point " + ScriptRuntime.toString(arg));
```

Comment:
I think the error messages start usually with an uppercase letter

Reply:
u r right.